platforms/osx/build_framework.py

```diff
@@ -54,13 +55,16 @@ def getInfoPlist(self, builddirs):
 
     os.environ['MACOSX_DEPLOYMENT_TARGET'] = args.macosx_deployment_target
     print('Using MACOSX_DEPLOYMENT_TARGET=' + os.environ['MACOSX_DEPLOYMENT_TARGET'])
+    archs = args.archs.split(',')
+    print('Using ARCHS=' + str(archs))
+
     if args.legacy_build:
         args.framework_name = "opencv2"
         if not "objc" in args.without:
             args.without.append("objc")
 
     b = OSXBuilder(args.opencv, args.contrib, False, False, args.without, args.disable, args.enablenonfree,
         [
-            (["x86_64"], "MacOSX")
+            (archs, "MacOSX")
```

Comment:
Please use this form here:
```
("x86_64", "MacOSX"),
("arm64", "MacOSX"),
```
(as result)

Reply:
OSXBuilder shares code with the iOSBuilder so we need to pass the parameters in a way that is consistent with how it is done on iOS:
```
[
    (["armv7, "armv7s", "arm64"], "iPhoneOS"),
    (["x86_64", "i386"], "iPhoneSimulator"),
]
```

Reply:
Right.
I have confused by `arch = archs[0]` line in `makeCMakeCmd()`.

For each arch we should have own dedicated CMake pass.
It is impossible to "merge" them - current CMake scripts are targeted to single configuration at once.

Reply:
> I have confused by arch = archs[0] line in makeCMakeCmd()

OK I understand what you mean - I can fix that